aries_cloudagent/multitenant/tests/test_askar_profile_manager.py

```diff
@@ -0,0 +1,118 @@
+from unittest.mock import MagicMock
+
+from asynctest import TestCase as AsyncTestCase
+from asynctest import mock as async_mock
+
+from ...config.injection_context import InjectionContext
+from ...core.in_memory import InMemoryProfile
+from ...messaging.responder import BaseResponder
+from ...wallet.models.wallet_record import WalletRecord
+from ..askar_profile_manager import AskarProfileMultitenantManager
+
+class TestAskarProfileMultitenantManager(AsyncTestCase):
+    DEFAULT_MULTIENANT_WALLET_NAME = "multitenant_sub_wallet"
+
+    async def setUp(self):
+        self.profile = InMemoryProfile.test_profile()
+        self.context = self.profile.context
+
+        self.responder = async_mock.CoroutineMock(send=async_mock.CoroutineMock())
+        self.context.injector.bind_instance(BaseResponder, self.responder)
+
+        self.manager = AskarProfileMultitenantManager(self.profile)
+
+    async def test_get_wallet_profile_should_open_store_and_return_profile_with_wallet_context(self):
+        askar_profile_mock_name = "AskarProfile"
+        wallet_record = WalletRecord(
+            wallet_id="test",
+            settings={
+                "wallet.recreate": True,
+                "wallet.seed": "test_seed",
+                "wallet.name": "test_name",
+                "wallet.type": "test_type",
+                "wallet.rekey": "test_rekey",
+                "mediation.open": True,
+                "mediation.invite": "http://invite.com",
+                "mediation.default_id": "24a96ef5",
+                "mediation.clear": True,
+            },
+        )
+
+        with async_mock.patch(
+            "aries_cloudagent.multitenant.askar_profile_manager.wallet_config"
+        ) as wallet_config:
+            with async_mock.patch(
+                "aries_cloudagent.multitenant.askar_profile_manager.AskarProfile"
+            ) as AskarProfile:
+                sub_wallet_profile_context = InjectionContext()
+                sub_wallet_profile = AskarProfile(None, None)
+                sub_wallet_profile.context.copy.return_value = sub_wallet_profile_context
+                def side_effect(context, provision):
+                    sub_wallet_profile.name = askar_profile_mock_name
+                    return sub_wallet_profile, None
+                wallet_config.side_effect = side_effect
+
+                profile = await self.manager.get_wallet_profile(self.profile.context, wallet_record)
+
+                assert profile.name == askar_profile_mock_name
+                wallet_config.assert_called_once()
```

Comment:
Can this be asserted with params?

Reply:
Params are asserted in the next lines because this assertion only works on object references